### PR TITLE
Update usage message in makecpt

### PR DESCRIPTION
**Pull request by PaulWessel**
As per #5341.  Also cleaned up **-F** in **grd2cpt** the same way.

### CODE REVIEW

File: src/makecpt.c [hunk 1]
```diff
@@ -144,61 +144,73 @@ static int usage (struct GMTAPI_CTRL *API, int level) {
...
+		name, H_OPT, GMT_V_OPT, GMT_bi_OPT, GMT_di_OPT, GMT_i_OPT, GMT_ho_OPT, GMT_PAR_OPT);
 
 	if (level == GMT_SYNOPSIS) return (GMT_MODULE_SYNOPSIS);
 
-	GMT_Message (API, GMT_TIME_NONE, "\n  OPTIONAL ARGUMENTS:\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-A Set constant transparency for all colors; append +a to also include back-, for-, and nan-colors [0]\n");
+	GMT_Message (API, GMT_TIME_NONE, "  OPTIONAL ARGUMENTS:\n");
+	GMT_Usage (API, 1, "\n-A<transparency>[+a]");
+	GMT_Usage (API, -2, "Set constant transparency for all colors; append +a to also include back-, for-, and nan-colors [0].");
 	if (gmt_list_cpt (API->GMT, 'C')) return (GMT_CPT_READ_ERROR);	/* Display list of available color tables */
```
Comment (meghanrjones) [inline src/makecpt.c:157]: Could the gmt_message calls in gmt_list_cpt be updated to GMT_Usage? A few have been, but not all.

Comment (PaulWessel) [reply src/makecpt.c:157]: OK, did that - should look fine now regardless of terminal width.

File: src/makecpt.c [hunk 2]
```diff
@@ -144,61 +144,73 @@ static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	const char *H_OPT = (API->GMT->current.setting.run_mode == GMT_MODERN) ? " [-H]" : "";
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [-A<transparency>[+a]] [-C<cpt>|colors] [-D[i|o]] [-E<nlevels>] [-F[R|r|h|c][+c[<label>]][+k<keys>]] [-G<zlo>/<zhi>]%s\n", name, H_OPT);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-I[c][z]] [-M] [-N] [-Q] [-S<mode>] [-T<min>/<max>[/<inc>[+b|i|l|n]] | -T<table> | -T<z1,z2,...zn>] [%s] [-W[w]]\n\t[-Z] [%s] [%s] [%s]\n\t[%s] [%s]\n\n",
-		GMT_V_OPT, GMT_bi_OPT, GMT_di_OPT, GMT_i_OPT, GMT_ho_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s [-A<transparency>[+a]] [-C<cpt>|colors] [-D[i|o]] [-E<nlevels>] "
```
Comment (meghanrjones) [inline src/makecpt.c:147]:
```suggestion
	GMT_Usage (API, 0, "usage: %s [-A<transparency>[+a]] [-C<cpt>|colors] [-D[i|o]] [-E[<nlevels>]] "
```

File: src/makecpt.c [hunk 3]
```diff
@@ -144,61 +144,73 @@ static int usage (struct GMTAPI_CTRL *API, int level) {
...
-	GMT_Message (API, GMT_TIME_NONE, "\t   Append +b for log2 spacing in integer <inc> or +l for log10 spacing via <inc> = 1,2,3.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Append +n to indicate <inc> is the number of color boundaries to produce instead.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Alternatively, append +i to indicate <inc> is the reciprocal of desired <inc> (e.g., 3 for 0.3333.....).\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   For absolute time series, append a valid time unit (%s) to the increment.\n", GMT_TIME_UNITS_DISPLAY);
-	GMT_Message (API, GMT_TIME_NONE, "\t   Alternatively, give a file with color boundaries in the first column, or a comma-separate list of values.\n");
+	GMT_Usage (API, 1, "\n-D[i|o]");
+	GMT_Usage (API, -2, "Set back- and foreground color to match the bottom/top limits "
+		"in the output CPT [Default (-D or -Do) uses the output color table]. Append i "
+		"to match the bottom/top values in the input CPT instead.");
+	GMT_Usage (API, 1, "\n-E<nlevels>");
```
Comment (meghanrjones) [inline src/makecpt.c:162]:
```suggestion
	GMT_Usage (API, 1, "\n-E[<nlevels>]");
```

File: src/makecpt.c [hunk 4]
```diff
@@ -144,61 +144,73 @@ static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	const char *H_OPT = (API->GMT->current.setting.run_mode == GMT_MODERN) ? " [-H]" : "";
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [-A<transparency>[+a]] [-C<cpt>|colors] [-D[i|o]] [-E<nlevels>] [-F[R|r|h|c][+c[<label>]][+k<keys>]] [-G<zlo>/<zhi>]%s\n", name, H_OPT);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-I[c][z]] [-M] [-N] [-Q] [-S<mode>] [-T<min>/<max>[/<inc>[+b|i|l|n]] | -T<table> | -T<z1,z2,...zn>] [%s] [-W[w]]\n\t[-Z] [%s] [%s] [%s]\n\t[%s] [%s]\n\n",
-		GMT_V_OPT, GMT_bi_OPT, GMT_di_OPT, GMT_i_OPT, GMT_ho_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s [-A<transparency>[+a]] [-C<cpt>|colors] [-D[i|o]] [-E<nlevels>] "
+		"[-F[R|r|h|c][+c[<label>]][+k<keys>]] [-G<zlo>/<zhi>]%s [-I[c][z]] [-M] [-N] [-Q] [-S<mode>] "
+		"[-T<min>/<max>[/<inc>[+b|i|l|n]] | -T<table> | -T<z1,z2,...zn>] [%s] [-W[w]] [-Z] [%s] [%s] [%s] [%s] [%s]\n",
+		name, H_OPT, GMT_V_OPT, GMT_bi_OPT, GMT_di_OPT, GMT_i_OPT, GMT_ho_OPT, GMT_PAR_OPT);
```
Comment (meghanrjones) [inline src/makecpt.c:150]:
```suggestion
		name, H_OPT, GMT_V_OPT, GMT_bi_OPT, GMT_di_OPT, GMT_ho_OPT, GMT_i_OPT, GMT_PAR_OPT);
```

File: src/makecpt.c [hunk 5]
```diff
@@ -144,61 +144,73 @@ static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	const char *H_OPT = (API->GMT->current.setting.run_mode == GMT_MODERN) ? " [-H]" : "";
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [-A<transparency>[+a]] [-C<cpt>|colors] [-D[i|o]] [-E<nlevels>] [-F[R|r|h|c][+c[<label>]][+k<keys>]] [-G<zlo>/<zhi>]%s\n", name, H_OPT);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-I[c][z]] [-M] [-N] [-Q] [-S<mode>] [-T<min>/<max>[/<inc>[+b|i|l|n]] | -T<table> | -T<z1,z2,...zn>] [%s] [-W[w]]\n\t[-Z] [%s] [%s] [%s]\n\t[%s] [%s]\n\n",
-		GMT_V_OPT, GMT_bi_OPT, GMT_di_OPT, GMT_i_OPT, GMT_ho_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s [-A<transparency>[+a]] [-C<cpt>|colors] [-D[i|o]] [-E<nlevels>] "
+		"[-F[R|r|h|c][+c[<label>]][+k<keys>]] [-G<zlo>/<zhi>]%s [-I[c][z]] [-M] [-N] [-Q] [-S<mode>] "
+		"[-T<min>/<max>[/<inc>[+b|i|l|n]] | -T<table> | -T<z1,z2,...zn>] [%s] [-W[w]] [-Z] [%s] [%s] [%s] [%s] [%s]\n",
```
Comment (meghanrjones) [inline src/makecpt.c:149]:
```suggestion
		"[-T<min>/<max>/<inc>[+b|i|l|n] | -T<table> | -T<z1,z2,...zn>] [%s] [-W[w]] [-Z] [%s] [%s] [%s] [%s] [%s]\n",
```

**Review by meghanrjones — APPROVED** [review]
Great!

File: src/makecpt.c (hunk 5 above)
Comment (PaulWessel) [reply src/makecpt.c:149]: Hm, not sure about that.  For continuous CPTs you would just say -T-300/600 as there are no specific intervals other than whatever happens in the master CPT.

**Comment by meghanrjones** [comment]
Not quite ready yet actually, breaks with really narrow terminal (<54 char)

File: src/makecpt.c (hunk 5 above)
Comment (meghanrjones) [reply src/makecpt.c:149]: OK, L203 needs to be updated to reflect the continuous CPT option.

**Comment by PaulWessel** [comment]
> Not quite ready yet actually, breaks with really narrow terminal (<54 char)

What breaks? Looks great on this end for tiny terminal widths...

**Comment by PaulWessel** [comment]
OK, I see. Hold on.

**Comment by PaulWessel** [comment]
Wil continue this later - gotta go..

**Comment by PaulWessel** [comment]
OK, good one.  For the rainbow we had this line

`"rainbow     : Rainbow: magenta-blue-cyan-green-yellow-red [C=HSV]"`

and the sub-function _gmtapi_split_words_ did not consider a hyphen as a place to break long words (...) so that list of color was a single word that exceeded the length of the terminal minus indent (19).  So it got stuck in an infinite loop until an array spilled over.  The fix here is two-fold:

1. First, allow a hyphen followed by a letter to be a valid place to break a word.  This fixes the problem above case.
2. Detect when this situation happens (word too long) and force it to be written anyway and move on.  This prevents a crash.

I tested this by replacing those hyphens with + symbols and it forced the output and moved on.

**Pull request merged**